packages/flame/lib/src/effects/opacity_effect.dart

```diff
@@ -0,0 +1,82 @@
+import 'dart:ui';
+
+import 'package:flutter/animation.dart';
+
+import '../components/mixins/has_paint.dart';
+import 'effects.dart';
+
+class OpacityEffect extends ComponentEffect<HasPaint> {
```

Comment:
I would create a `ColorEffect` that this one extends and then the color effect just lerps each color field separately (RGBA) from one `Color` to another `Color`, since `opacity` is a field in `Color`.

Reply:
I will governor a try, but I think that we need to have separate effects because iirc to change the color we need to use color blending

Reply:
You could use something like this:
```dart
Color.fromARGB(
      lerp(original.alpha, end.alpha, curveProgress),
      lerp(original.red, end.green, curveProgress),
      lerp(original.green, end.green, curveProgress),
      lerp(original.blue, end.blue,  );`
```

Reply:
Indeed can't use like that to tint the paint, I have added a `tint` method on the mixin and a color effect though